src/views/instances/new-dhis2/fields/name-input.tsx

```diff
@@ -22,13 +22,13 @@ const validateDnsLabel = (value = '') => {
     if (value.length > maxLenght) {
         return `Name can only contain ${maxLenght} characters`
     }
-    if (!startPattern.test(value)) {
+    if (!startPattern.test(value.charAt(0))) {
         return 'Name must start with a lowercase letter'
     }
-    if (!endPattern.test(value)) {
+    if (!endPattern.test(value.charAt(value.length - 1))) {
         return 'Name must end with a lowercase letter or a number'
     }
-    if (!interiorPattern.test(value)) {
+    if (!interiorPattern.test(value.substring(1, value.length - 1))) {
```

Comment:
For the first two cases it make sense to only look at the relevant part of the string, but not for the latter. All it does is complicating both the regular expression and the if statement.